community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -473,9 +436,22 @@ public long closeTransaction() throws TransactionFailureException
         }
     }
 
-    public boolean isClosing()
+    private void markTransactionAsClosing() throws TransactionFailureException
     {
-        return closing;
+        if ( !transactionStatus.closing() )
+        {
+            assertTransactionOpen();
+            if ( transactionStatus.isShutdown() )
+            {
+                throw new TransactionFailureException( Status.General.DatabaseUnavailable,
+                        "Transaction terminated since database is shutting down." );
```

Comment:
I wonder if this message is correct. A transaction is shut down in KTI#dispose(), which may be called for various reasons, one of them shutting down the database.